river/cluster/clustream.py

```diff
@@ -0,0 +1,363 @@
+            timestamp=self.time_stamp,
+            T=self.kernel_radius_factor,
+            M=self.max_kernels,
+        )
+
+        return self
+
+    def get_micro_clustering_result(self):
+
+        if not self.initialized:
+            return {}
+        res = {
+            i: ClustreamKernel(
+                cluster=self.kernels[i], T=self.kernel_radius_factor, M=self.max_kernels
+            )
+            for i in range(len(self.kernels))
+        }
+        return res
+
+    def learn_predict_one(self, x, sample_weight=None):
+        """
+        Compute cluster centers and predict cluster index for each sample.
+
+        Convenience method; equivalent to calling learn_one(x) followed by predict_one(x).
+
+        Parameters
+        ----------
+        x
+            Instance attributes
+
+        sample_weight
+            Instance weights. If not provided, uniform weights are assumed
+
+        Returns
+        -------
+        y
+            Integer
+            Cluster label
+        """
+
+        self.learn_one(x, sample_weight)
+
+        y = self.predict_one(x)
+
+        return y
+
+    @staticmethod
+    def _get_closest_kernel(x, micro_clusters):
+        """
+
+        Parameters
+        ----------
+        x
+            A dictionary with n items (features).
+            Instance attributes. Due to the limitations of the dictionary, x on default is considered as one instance
+
+        micro_clusters
+            Dictionary like
+            Instance weight. If not provided, uniform weights are assumed
+
+        Returns
+        -------
+        closest_kernel_index : int
+            Index of closest kernel to the given instance
+
+        min_distance: float
+            distance between closest kernel to the given instance
+        """
+
+        min_distance = sys.float_info.max
+        closest_kernel_index = -1
+        for i, micro_cluster in micro_clusters.items():
+            distance = 0
+            for j in range(len(x)):
+                distance += (micro_cluster.center[j] - x[j]) * (
+                    micro_cluster.center[j] - x[j]
+                )
+            distance = math.sqrt(distance)
+            if distance < min_distance:
+                min_distance = distance
+                closest_kernel_index = i
+        return closest_kernel_index, min_distance
+
+    @staticmethod
+    def _distance(point_a, point_b):
+        distance = 0.0
+        for i in range(len(point_a)):
+            d = point_a[i] - point_b[i]
+            distance += d * d
+        return math.sqrt(distance)
+
+    def predict_one(self, x):
+        """Predict cluster index for each sample.
+
+        Convenience method; equivalent to calling partial_fit(x) followed by predict(x).
+
+        Parameters
+        ----------
+        x
+            A dictionary with n items
+            Instance attributes
+
+        Returns
+        -------
+        label
+            Integer
```

Comment:
Remove types as they are colleced from the signature.

Reply:
Removed.